Forge/build.gradle

```diff
@@ -63,23 +109,43 @@ minecraft {
     }
 }
 
-sourceSets.main.resources.srcDir 'src/generated/resources'
+//createMcpToSrg.finalizedBy('genIntellijRuns')
+
+sourceSets.main.resources { srcDir 'src/generated/resources' }
 
 dependencies {
     minecraft "net.minecraftforge:forge:${minecraft_version}-${forge_version}"
-    compileOnly project(":Common")
-}
 
-tasks.withType(JavaCompile) {
-    source(project(":Common").sourceSets.main.allSource)
+    annotationProcessor "org.spongepowered:mixin:${mixin_version}:processor"
+
+    implementation project(":Common")
 }
 
 processResources {
     from project(":Common").sourceSets.main.resources
 }
 
+tasks.withType(JavaCompile).configureEach {
+    it.source(project(":Common").sourceSets.main.allSource)
+}
+
+sourcesJar {
+    if (Boolean.parseBoolean(common_add_to_sources)) {
+        from project(":Common").sourceSets.main.allSource
+    }
+}
+
+javadoc {
+    if (Boolean.parseBoolean(common_add_to_javadoc)) {
+        source(project(":Common").sourceSets.main.allJava)
+    }
+}
```

Comment:
Someone might want only modloader-specific code in corresponding sources/javadoc, and add common sources to a separate .jar.

Reply:
This will cause needless confusion, if the common source is not included in the javadocs and source files, people pulling just the fabric version of the mod from maven, won't have any documentation for common code for example.

There should be no reason for the sources jar to be different than the built jar in terms of content covered.